src/tiledquickplugin/maploader.cpp

```diff
@@ -46,7 +46,9 @@ void MapLoader::setSource(const QUrl &source)
 
     Tiled::MapReader mapReader;
 
-    Tiled::Map *map = mapReader.readMap(source.toLocalFile());
+    const QString qrcScheme = QStringLiteral("qrc");
+    const bool isQrcScheme = source.scheme() == qrcScheme;
+    Tiled::Map *map = mapReader.readMap(isQrcScheme ? source.toString().mid(qrcScheme.length()) : source.toLocalFile());
```

Comment:
Right, I'd include a copy of `urlToLocalFileOrQrc` then and use that. It's unfortunate the QQmlFile class is `\internal`, so we shouldn't use it directly even though it's in public headers.